CI/unit_tests/memory_manager/test_memory_manager.py

```diff
@@ -0,0 +1,248 @@
+"""
+MDSuite: A Zincwarecode package.
+License
+-------
+This program and the accompanying materials are made available under the terms
+of the Eclipse Public License v2.0 which accompanies this distribution, and is
+available at https://www.eclipse.org/legal/epl-v20.html
+SPDX-License-Identifier: EPL-2.0
+Copyright Contributors to the Zincwarecode Project.
+Contact Information
+-------------------
+email: zincwarecode@gmail.com
+github: https://github.com/zincware
+web: https://zincwarecode.com/
+Citation
+--------
+If you use this module please cite us with:
+Summary
+-------
+Test for the memory manager module.
+"""
+import unittest
+
+import numpy as np
+
+from mdsuite.memory_management.memory_manager import MemoryManager
+
+
+class TestDatabase:
+    """
+    Test database for the unit testing.
+    """
+
+    def __init__(self, data_size: int = 500, rows: int = 10, columns: int = 10):
+        """
+        Constructor for the test database.
+        """
+        self.data_size = data_size
+        self.rows = rows
+        self.columns = columns
+
+    def get_data_size(self, item, system: bool):
+        """
+        Compute the size of a dataset.
+
+        Returns
+        -------
+
+        """
+        return self.rows, self.columns, self.data_size
+
+
+class TestMemoryManager(unittest.TestCase):
+    """
+    Test the memory manager module.
+    """
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        """
+        Prepare the class.
+
+        Returns
+        -------
+
+        """
+        cls.memory_manager = MemoryManager()
+
+    def test_linear_scale_function(self):
+        """
+        Test the linear scale selection.
+
+        Notes
+        -----
+        Each test will check that the function parameters are return correctly and
+        that the function is called correctly and returns proper values.
+        """
+        # Test linear function
+        scale_function = {"linear": {"scale_factor": 2}}
+        function, parameters = self.memory_manager._select_scale_function(
+            scale_function
+        )
+        self.assertEqual(parameters["scale_factor"], 2)
+        self.assertEqual(function(10, **parameters), 20)
+
+    def test_log_linear_scale_function(self):
+        """
+        Test the log-linear scale selection.
+
+        Notes
+        -----
+        Each test will check that the function parameters are return correctly and
+        that the function is called correctly and returns proper values.
+        """
+        # Test log-linear function
+        scale_function = {"log-linear": {"scale_factor": 2}}
```

Comment:
For me to understand this correctly, we do have the type of the scaling, e.g. `log-linear` and then the `scaling_factor` is in addition to that so `f(x) = scaling * x` for linear

Reply:
so the memory goes through the function and then is scaled by this factor, for log-linear:

```
memory_used = scaling * memory * log(memory)
```